backend/apps/matching/admin.py

```diff
@@ -42,3 +42,15 @@ def new_func(request, p_type, *args, **kwargs):
     if function:
         return actual_decorator(function)
     return actual_decorator
+
+
+def no_login_required(function=None, login_url="/"):
```

Comment:
Maybe rather this?
```suggestion
def logged_in_not_permitted(function=None, login_url="/"):
```
`no login required` suggests it is not required, not that it is not allowed...